### DISCUSSION TITLE
Consumer fails to consume messages

### DISCUSSION THREAD

**Discussion by shawn-happy** [apache/pulsar#20628] [Q&A]
有个pulsar的问题，需要问问大家，看看大家有没有排查思路。现象是6月19号中午12点多的时候，发布了新版本。
其中ETL任务正在跑着，可能由于当时ETL代码正在更新，consumer消费完消息正在发送ack给pulsar broker，然后ETL任务重启了。重启完了以后，这个ETL任务就一直消费不了pulsar里的消息，导致现在积压了2万多条消息。写数据写的进去，但是读数据读不了。
尝试过的换一个subscribtion。这里通过ETL换过，也自己写了一个程序在上面测试，都不行。
查看pulsar日志也没有发生报错。 
怀疑是不是因为没有完全ack掉，topic有个类似于锁一样的机制，有个标志位，这个标志位存在，就不能继续消费，需要保证状态一致性？

<img width="960" alt="image" src="https://github.com/apache/pulsar/assets/34005806/dd8397c5-436c-4db0-a780-940375de5cdf">

![image](https://github.com/apache/pulsar/assets/34005806/3cfc29a7-488c-420a-847e-d5a767a555cb)

后面的程序是个demo，换了subscribtion，也还是消费不了，上面的图片之所以没有test-sub主要是因为我看消费不了就先unsubscribe了

**Comment by tisonkun** [comment]
@shawn-happy could you please translate the question in English so that other community members can help? Note that the default language in the Pulsar community is English and we don't establish a Chinese-specific forum (yet).
